### PR TITLE
Warn if ASA is used in combination with events

### PR DESCRIPTION
**Pull request by dweindl**
#18

### CODE REVIEW

**Review by FFroehlich — COMMENTED** [review]
‚ne‘ counts discontinuities in the rhs and x. Discontinuities in the rhs are fine with adjoint sensitivity analysis as long as the trigger function is not parameter dependent.

**Comment by dweindl** [comment]
> ‚ne‘ counts discontinuities in the rhs and x. Discontinuities in the rhs are fine with adjoint sensitivity analysis as long as the trigger function is not parameter dependent.

Ok, thanks. As this is not apparent from the generated model module, and i don't feel like storing it there - how do we want to handle that?

1) change the message and live with the warnings for non-parameter-dependent trigger functions
2) rather warn once during model import 
3) leave it as it was

**Comment by FFroehlich** [comment]
> > ‚ne‘ counts discontinuities in the rhs and x. Discontinuities in the rhs are fine with adjoint sensitivity analysis as long as the trigger function is not parameter dependent.
> 
> Ok, thanks. As this is not apparent from the generated model module, and i don't feel like storing it there - how do we want to handle that?
> 
> 1. change the message and live with the warnings for non-parameter-dependent trigger functions
> 2. rather warn once during model import
> 3. leave it as it was before (no warning)

Hmm rephrase and state that it might not apply to the model at hand?

File: python/sdist/amici/swig_wrappers.py [hunk 1]
```diff
@@ -102,6 +104,18 @@ def runAmiciSimulation(
...
+    if (
+        model.ne > 0
+        and solver.getSensitivityMethod()
+        == amici_swig.SensitivityMethod.adjoint
+        and solver.getSensitivityOrder() == amici_swig.SensitivityOrder.first
+    ):
+        warnings.warn(
+            "Adjoint sensitivity analysis for models with events with parameter-dependent trigger functions has not been thoroughly tested. "
+            "Sensitivities might be wrong. Tracked at https://github.com/AMICI-dev/AMICI/issues/18. "
+            "If your model does not have parameter-dependent trigger functions, you can safely ignore this message."
```
Comment (FFroehlich) [inline python/sdist/amici/swig_wrappers.py:116]:
```suggestion
            "Adjoint sensitivity analysis may work if the location of the discontinuity is not parameter-dependent, but we still recommend testing accuracy of gradients."
```

File: python/sdist/amici/swig_wrappers.py [hunk 2]
```diff
@@ -152,6 +166,18 @@ def runAmiciSimulations(
...
     :returns: list of simulation results
     """
+    if (
+        model.ne > 0
+        and solver.getSensitivityMethod()
+        == amici_swig.SensitivityMethod.adjoint
+        and solver.getSensitivityOrder() == amici_swig.SensitivityOrder.first
+    ):
+        warnings.warn(
+            "Adjoint sensitivity analysis for models with events with parameter-dependent trigger functions has not been thoroughly tested. "
```
Comment (FFroehlich) [inline python/sdist/amici/swig_wrappers.py:176]:
```suggestion
            "Adjoint sensitivity analysis for models with discontinuous right hand sides (events/piecewise functions) has not been thoroughly tested. "
```

File: python/sdist/amici/swig_wrappers.py [hunk 3]
```diff
@@ -152,6 +166,18 @@ def runAmiciSimulations(
...
+            "Sensitivities might be wrong. Tracked at https://github.com/AMICI-dev/AMICI/issues/18. "
+            "If your model does not have parameter-dependent trigger functions, you can safely ignore this message."
```
Comment (FFroehlich) [inline python/sdist/amici/swig_wrappers.py:178]:
```suggestion
            "Adjoint sensitivity analysis may work if the location of the discontinuity is not parameter-dependent, but we still recommend testing accuracy of gradients."
```

**Review by FFroehlich — CHANGES_REQUESTED** [review]

**Review by FFroehlich — APPROVED** [review]

**Pull request merged**